drivers/display/ili9340/ili9340.c

```diff
@@ -0,0 +1,181 @@
+static void ili9340_set_mem_area(struct ili9340_data *data, const u16_t x,
+				 const u16_t y, const u16_t w, const u16_t h);
+
+int ili9340_init(struct device *dev)
+{
+	SYS_LOG_DBG("Initializing display driver");
+
+	struct ili9340_data *data = (struct ili9340_data *)dev->driver_data;
+
+	data->spi = device_get_binding(CONFIG_ILI9340_SPI_DEV_NAME);
+	if (data->spi == NULL) {
+		SYS_LOG_ERR("Could not get SPI device for ILI9340");
+		return -EPERM;
+	}
+
+	data->reset_gpio =
+	    device_get_binding(CONFIG_ILI9340_RESET_GPIO_PORT_NAME);
+	if (data->reset_gpio == NULL) {
+		SYS_LOG_ERR("Could not get GPIO port for ILI9340 reset");
+		return -EPERM;
+	}
+	gpio_pin_configure(data->reset_gpio, CONFIG_ILI9340_RESET_PIN,
+			   GPIO_DIR_OUT);
+
+	data->command_data_gpio =
+	    device_get_binding(CONFIG_ILI9340_CMD_DATA_GPIO_PORT_NAME);
+	if (data->command_data_gpio == NULL) {
+		SYS_LOG_ERR("Could not get GPIO port for ILI9340 command/data");
+		return -EPERM;
+	}
+	gpio_pin_configure(data->command_data_gpio, CONFIG_ILI9340_CMD_DATA_PIN,
+			   GPIO_DIR_OUT);
+
+	SYS_LOG_DBG("Resetting display driver");
+	gpio_pin_write(data->reset_gpio, CONFIG_ILI9340_RESET_PIN, 1);
+	k_sleep(1);
+	gpio_pin_write(data->reset_gpio, CONFIG_ILI9340_RESET_PIN, 0);
+	k_sleep(1);
+	gpio_pin_write(data->reset_gpio, CONFIG_ILI9340_RESET_PIN, 1);
+	k_sleep(5);
+
+	SYS_LOG_DBG("Initializing LCD");
+	ili9340_lcd_init(data);
+
+	SYS_LOG_DBG("Exiting sleep mode");
+	ili9340_exit_sleep(data);
+
+	/* device_get_binding checks if driver_api is not zero before checking
+	 * device name.
+	 * So just set driver_api to 1 else the function call will fail
+	 */
+	dev->driver_api = (void *)1;
+
+	return 0;
+}
+
+void ili9340_write_pixel(const struct device *dev, const u16_t x, const u16_t y,
+			 const u8_t r, const u8_t g, const u8_t b)
+{
+	SYS_LOG_DBG("Writing pixel");
+	u8_t rgb_data[] = {r, g, b};
+	ili9340_write_bitmap(dev, x, y, 1, 1, &rgb_data[0]);
+}
+
+void ili9340_write_bitmap(const struct device *dev, const u16_t x,
+			  const u16_t y, const u16_t w, const u16_t h,
+			  const u8_t *rgb_data)
+{
+	SYS_LOG_DBG("Writing bitmap");
+	struct ili9340_data *data = (struct ili9340_data *)dev->driver_data;
+	ili9340_set_mem_area(data, x, y, w, h);
+	ili9340_transceive(data, ILI9340_CMD_MEM_WRITE, (void *)rgb_data,
+			   3 * w * h, NULL, 0);
+}
+
+void ili9340_display_on(struct device *dev)
+{
+	SYS_LOG_DBG("Turning display on");
+	struct ili9340_data *data = (struct ili9340_data *)dev->driver_data;
+	ili9340_transceive(data, ILI9340_CMD_DISPLAY_ON, NULL, 0, NULL, 0);
+}
+
+void ili9340_display_off(struct device *dev)
+{
+	SYS_LOG_DBG("Turning display off");
+	struct ili9340_data *data = (struct ili9340_data *)dev->driver_data;
+	ili9340_transceive(data, ILI9340_CMD_DISPLAY_OFF, NULL, 0, NULL, 0);
+}
+
+void ili9340_transceive(struct ili9340_data *data, u8_t cmd, void *tx_data,
+			size_t tx_len, void *rx_data, size_t rx_len)
+{
+	struct spi_config spi_config = {
```

Comment:
Doing so won't perform nicely. Here you'll force the SPI controller to reconfigure itself on each ili9340_transceive() call, as this spi_config pointer might vary (since it's allocated on the stack).

Better include your struct spi_config into your struct ili9340_data (remove the struct device spi pointer there, as spi_config has one). And initialize the struct only once. of course.

I see that you have 2 different frequencies for tx and rx. This changes would indeed need a different config.
However, as far as I can tell, as you never read from the display... just scrap the rx part. (remove the parametrs from ili9340_transceive() as well then)